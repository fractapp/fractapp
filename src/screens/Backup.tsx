import React, {useContext, useEffect} from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  Text,
  FlatList,
  Image,
  TouchableOpacity,
} from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Dialog from 'storage/Dialog';
import BackupUtils from 'utils/backup';
import db from 'utils/db';
import backupUtil from 'utils/backup';

export const Backup = ({navigation, route}: {navigation: any; route: any}) => {
  const dialogContext = useContext(Dialog.Context);

  const onSuccess = () => {
    navigation.reset({
      index: 0,
      routes: [{name: 'Home'}],
    });
    dialogContext.dispatch(Dialog.close());
  };

  useEffect(() => {
    if (route.params?.isSuccess == null) {
      return;
    }

    const dir =
      route.params.type == BackupUtils.BackupType.File
        ? 'Downloads'
        : backupUtil.GoogleDriveFolder;

    dialogContext.dispatch(
      Dialog.open(
        'Success save wallet',
        `If you lose access to file then you will not be able to restore access to the wallet. File "${route.params?.fileName}.json" saved in "${dir}" directory`,
        onSuccess,
      ),
    );
  }, [route.params?.isSuccess]);

  const backupFile = async () => {
    const seed = await db.getSeed();
    await BackupUtils.backupFile(
      () =>
        navigation.navigate('WalletFileBackup', {
          seed: seed?.split(' '),
          type: BackupUtils.BackupType.File,
          callerScreen: 'Backup',
        }),
      () =>
        dialogContext.dispatch(
          Dialog.open(
            'Open settings',
            'If you want to save a file then open the application settings and give it access to the storage.',
            () => dialogContext.dispatch(Dialog.close()),
          ),
        ),
    );
  };
  const backupGoogleDrive = async () => {
    const seed = await db.getSeed();
    await BackupUtils.backupGoogleDrive(() =>
      navigation.navigate('WalletFileBackup', {
        seed: seed?.split(' '),
        type: BackupUtils.BackupType.GoogleDrive,
        callerScreen: 'Backup',
      }),
    );
  };
  const backupSeed = async () => {
    const seed = await db.getSeed();
    navigation.navigate('SaveSeed', {
      seed: seed?.split(' '),
      onSuccess: () =>
        navigation.reset({
          index: 0,
          routes: [{name: 'Home'}],
        }),
    });
  };

  const menuItems = [
    {
      img: (
        <Image
          source={require('assets/img/google-drive.png')}
          style={{width: 32, height: 32}}
        />
      ),
      title: 'Google disk',
      onClick: () => backupGoogleDrive(),
    },
    {
      img: <MaterialIcons name="folder-open" size={32} color="#888888" />,
      title: 'File to phone',
      onClick: () => backupFile(),
    },
    {
      img: (
        <MaterialCommunityIcons name="content-copy" size={32} color="#888888" />
      ),
      title: 'Backup seed',
      onClick: () => backupSeed(),
    },
  ];
  const renderItem = ({item}: {item: any}) => (
    <TouchableOpacity style={styles.menuItem} onPress={item.onClick}>
      {item.img}
      <Text style={styles.menuTitle}>{item.title}</Text>
    </TouchableOpacity>
  );

  return (
    <View style={styles.profile}>
      <ScrollView showsVerticalScrollIndicator={false} style={{width: '100%'}}>
        <View style={{alignItems: 'center', justifyContent: 'center'}}>
          <Text style={styles.description}>
            Save your wallet in a safe place. If you lose your wallet, you
            cannot restore access to it.
          </Text>
          <FlatList
            style={styles.menu}
            ItemSeparatorComponent={() => <View style={styles.dividingLine} />}
            data={menuItems}
            renderItem={renderItem}
            keyExtractor={(item) => item.title}
          />
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  profile: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dividingLine: {
    alignSelf: 'center',
    width: '100%',
    borderBottomWidth: 1,
    borderBottomColor: '#DADADA',
  },
  description: {
    width: '90%',
    marginTop: 10,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  menu: {
    flex: 1,
    width: '88%',
    marginTop: 20,
  },
  menuItem: {
    flexDirection: 'row',
    paddingTop: 10,
    paddingBottom: 10,
  },
  menuTitle: {
    marginLeft: 10,
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    color: 'black',
    alignSelf: 'center',
  },
});
