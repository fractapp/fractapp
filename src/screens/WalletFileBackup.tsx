import React, {useState, useContext, useEffect} from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {BlueButton, TextInput, PasswordInput, Loader} from 'components';
import db from 'storage/DB';
import backupUtil from 'utils/backup';
import Dialog from 'storage/Dialog';
import BackupUtils from 'utils/backup';
import GlobalStore from 'storage/Global';

const minPasswordLength = 6;

export const WalletFileBackup = ({
  navigation,
  route,
}: {
  navigation: any;
  route: any;
}) => {
  const dialogContext = useContext(Dialog.Context);
  const authContext = useContext(GlobalStore.Context);

  const [fileName, setFilename] = useState<string>(backupUtil.randomFilename());
  const [password, setPassword] = useState<string>('');
  const [confirmPassword, setConfirmPassword] = useState<string>('');
  const [isLoading, setLoading] = useState<boolean>(false);

  const seed: string = route.params.seed.join(' ');
  const type: backupUtil.BackupType = route.params.type;
  const isNewAccount: backupUtil.BackupType = route.params.isNewAccount;

  const startBackup = async () => {
    setLoading(true);
  };

  useEffect(() => {
    if (!isLoading) {
      return;
    }

    backupUtil.backup(seed, password, fileName, type).then(async () => {
      setLoading(false);

      await db.createAccounts(seed);
      const dir =
        route.params.type == BackupUtils.BackupType.File
          ? 'Downloads'
          : backupUtil.GoogleDriveFolder;

      dialogContext.dispatch(
        Dialog.open(
          'Success save wallet',
          `If you lose access to file then you will not be able to restore access to the wallet. File "${fileName}.json" saved in "${dir}" directory`,
          async () => {
            await dialogContext.dispatch(Dialog.close());

            if (isNewAccount) {
              await authContext.dispatch(GlobalStore.signInLocal());
            }

            navigation.reset({
              index: 0,
              routes: [{name: 'Home'}],
            });
          },
        ),
      );
    });
  }, [isLoading]);

  const renderButtonOrError = () => {
    if (confirmPassword == '' || password == '') {
      return null;
    } else if (
      password.length < minPasswordLength &&
      password != '' &&
      confirmPassword != ''
    ) {
      return (
        <Text style={styles.error}>
          Minimum password length is 6 characters
        </Text>
      );
    } else if (password != confirmPassword) {
      return <Text style={styles.error}>Password do not match</Text>;
    }

    return (
      <View style={{width: '80%', position: 'absolute', bottom: 40}}>
        <BlueButton text={'Encrypt'} height={50} onPress={startBackup} />
      </View>
    );
  };

  if (isLoading) {
    return <Loader />;
  }

  return (
    <View
      style={{
        flexDirection: 'column',
        flex: 1,
        alignItems: 'center',
      }}>
      <Text style={styles.title}>Wallet encryption</Text>
      <Text style={styles.description}>
        Enter the password to encrypt your wallet. Do not lose your password
        otherwise you will not be able to restore access.
      </Text>

      <View style={styles.fileName}>
        <TextInput
          onChangeText={(value: string) => setFilename(value)}
          placeholder={'File name'}
          defaultValue={fileName}
        />
      </View>
      <View style={styles.newPassword}>
        <PasswordInput
          onChangeText={(value: string) => setPassword(value)}
          placeholder={'Password'}
          defaultValue={password}
        />
      </View>
      <View style={styles.confirmPassword}>
        <PasswordInput
          onChangeText={(value: string) => setConfirmPassword(value)}
          placeholder={'Confirm password'}
          defaultValue={confirmPassword}
        />
      </View>
      {renderButtonOrError()}
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    marginTop: 80,
    fontSize: 25,
    fontFamily: 'Roboto-Regular',
    color: '#2AB2E2',
  },
  description: {
    width: '90%',
    marginTop: 40,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  fileName: {
    marginTop: 30,
    width: '90%',
  },
  newPassword: {
    marginTop: 20,
    width: '90%',
  },
  confirmPassword: {
    marginTop: 20,
    width: '90%',
  },
  error: {
    marginTop: 20,
    color: 'red',
    fontFamily: 'Roboto-Regular',
    fontSize: 15,
  },
});
