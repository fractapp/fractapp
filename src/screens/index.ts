export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
