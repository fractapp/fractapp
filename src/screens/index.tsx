export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'