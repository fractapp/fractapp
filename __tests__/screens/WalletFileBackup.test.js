import React, {useContext, useState} from 'react';
import {WalletFileBackup} from 'screens/WalletFileBackup';
import renderer from 'react-test-renderer';
import backupUtil from 'utils/backup';
import GlobalStore from 'storage/Global';
import DialogStore from 'storage/Dialog';
import {fireEvent, render} from '@testing-library/react-native';
import BackupUtils from 'utils/backup';
import DB from 'storage/DB';
import StringUtils from 'utils/string';
import {NativeModules} from 'react-native';
import googleUtils from 'utils/google';

jest.mock('storage/DB', () => ({
  createAccounts: jest.fn(),
}));
jest.mock('react-native-crypto', () => {});
jest.mock('react-native-fs', () => {});
jest.mock('react-native-share', () => {});
jest.mock('@polkadot/util-crypto', () => ({
  randomAsHex: jest.fn(),
}));
jest.mock('utils/google', () => ({
  getFileBackup: jest.fn(),
}));
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useContext: jest.fn(),
}));
jest.mock('utils/backup', () => ({
  getWalletsFromGoogle: jest.fn(),
  GoogleDriveFolder: 'fractapp',
  BackupType: {
    GoogleDrive: 12,
    File: 13,
  },
  checkPermissions: jest.fn(),
  backup: jest.fn(),
  getSeed: jest.fn(),
  randomFilename: jest.fn(() => '0xff00ff00'),
}));
jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

jest.mock('react-native', () => ({
  NativeModules: {
    PreventScreenshotModule: {
      forbid: jest.fn(() => new Promise.resolve({data: {}})),
    },
  },
}));
useState.mockImplementation((init) => [init, jest.fn()]);

const params = {
  seed: 'staff select toddler junior robot own paper sniff glare drive stay census'.split(
    ' ',
  ),
  type: backupUtil.BackupType.File,
  isNewAccount: true,
};

it('Test empty password', () => {
  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: jest.fn(),
  });
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });

  const tree = renderer
    .create(<WalletFileBackup route={{params: params}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test loading', () => {
  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => [true, jest.fn()]);

  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: jest.fn(),
  });

  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });

  const tree = renderer
    .create(<WalletFileBackup route={{params: params}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test minimum password length', () => {
  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => ['123', jest.fn()])
    .mockImplementationOnce((init) => ['123', jest.fn()]);

  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: jest.fn(),
  });

  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });

  const tree = renderer
    .create(<WalletFileBackup route={{params: params}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test password do not match', () => {
  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => ['123123', jest.fn()])
    .mockImplementationOnce((init) => ['123111', jest.fn()]);

  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: jest.fn(),
  });

  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });

  const tree = renderer
    .create(<WalletFileBackup route={{params: params}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test success password', () => {
  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => ['123123', jest.fn()])
    .mockImplementationOnce((init) => ['123123', jest.fn()]);

  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: jest.fn(),
  });

  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });

  const tree = renderer
    .create(<WalletFileBackup route={{params: params}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test click encrypt', () => {
  let isLoader = false;
  const setLoader = jest.fn((value) => (isLoader = value));

  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => ['Aa123123', jest.fn()])
    .mockImplementationOnce((init) => ['Aa123123', jest.fn()])
    .mockImplementationOnce((init) => [isLoader, setLoader]);

  const dialogDispatch = jest.fn();
  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: dialogDispatch,
  });

  const globalDispatch = jest.fn();
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: globalDispatch,
  });

  const mockNav = jest.fn();

  const component = render(
    <WalletFileBackup route={{params: params}} navigation={{reset: mockNav}} />,
  );
  fireEvent.press(component.getByText(StringUtils.texts.ConfirmBtnTitle));

  expect(setLoader).toBeCalledWith(true);
});

it('Test backup', async () => {
  const setLoading = jest.fn();
  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => ['Aa123123', jest.fn()])
    .mockImplementationOnce((init) => ['Aa123123', jest.fn()])
    .mockImplementationOnce((init) => [true, setLoading]);

  const dialogDispatch = jest.fn();
  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: dialogDispatch,
  });

  const globalDispatch = jest.fn();
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: globalDispatch,
  });

  backupUtil.backup.mockReturnValueOnce({
    isExist: false,
    isError: false,
  });
  backupUtil.getWalletsFromGoogle.mockReturnValueOnce({
    wallets: ['0xff00ff00.json'],
    ids: ['123123'],
  });
  googleUtils.getFileBackup.mockReturnValueOnce({
    seed: params.seed.join(' '),
    algorithm: 'algorithm',
  });
  backupUtil.getSeed.mockReturnValueOnce(params.seed.join(' '));

  const mockNav = jest.fn();

  const component = await render(
    <WalletFileBackup route={{params: params}} navigation={{reset: mockNav}} />,
  );

  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, 1000);
  }); //TODO

  expect(googleUtils.getFileBackup).toBeCalledWith('123123');
  expect(backupUtil.getSeed).toBeCalledWith(
    {
      seed: params.seed.join(' '),
      algorithm: 'algorithm',
    },
    'Aa123123',
  );
  expect(setLoading).toBeCalledWith(false);
  expect(BackupUtils.backup).toBeCalledWith(
    params.seed.join(' '),
    'Aa123123',
    '0xff00ff00',
    params.type,
  );
  expect(DB.createAccounts).toBeCalledWith(params.seed.join(' '));

  expect(mockNav).toBeCalledWith({
    index: 0,
    routes: [{name: 'Home'}],
  });
});
