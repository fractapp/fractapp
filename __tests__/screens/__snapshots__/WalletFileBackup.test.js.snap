// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test empty password 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
        "marginTop": 80,
      }
    }
  >
    Wallet encryption
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 40,
        "width": "90%",
      }
    }
  >
    Enter the password to encrypt your wallet. Do not lose your password otherwise you will not be able to restore access.
  </Text>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="name"
      defaultValue="fractapp-0xff00ff00"
      onChangeText={[Function]}
      placeholder="File name"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`Test loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
        "marginTop": 80,
      }
    }
  >
    Wallet encryption
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 40,
        "width": "90%",
      }
    }
  >
    Enter the password to encrypt your wallet. Do not lose your password otherwise you will not be able to restore access.
  </Text>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="name"
      defaultValue="fractapp-0xff00ff00"
      onChangeText={[Function]}
      placeholder="File name"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue={true}
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`Test minimum password length 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
        "marginTop": 80,
      }
    }
  >
    Wallet encryption
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 40,
        "width": "90%",
      }
    }
  >
    Enter the password to encrypt your wallet. Do not lose your password otherwise you will not be able to restore access.
  </Text>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="name"
      defaultValue="123"
      onChangeText={[Function]}
      placeholder="File name"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue="123"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`Test password do not match 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
        "marginTop": 80,
      }
    }
  >
    Wallet encryption
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 40,
        "width": "90%",
      }
    }
  >
    Enter the password to encrypt your wallet. Do not lose your password otherwise you will not be able to restore access.
  </Text>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="name"
      defaultValue="123123"
      onChangeText={[Function]}
      placeholder="File name"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue="222222"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`Test success password 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
        "marginTop": 80,
      }
    }
  >
    Wallet encryption
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 40,
        "width": "90%",
      }
    }
  >
    Enter the password to encrypt your wallet. Do not lose your password otherwise you will not be able to restore access.
  </Text>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="name"
      defaultValue="123123"
      onChangeText={[Function]}
      placeholder="File name"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue="123123"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="#BFBDBD"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
